recipes/soplex/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package CXX)
+set(CMAKE_CXX_STANDARD 14)
+
+find_package(soplex REQUIRED CONFIG)
```

These will be depreated soon soo...  lets try this 🤞 

```suggestion
find_package(soplex REQUIRED CONFIG)

# work around v1 legacy generators always having namespaces
if(TARGET soplex::soplex)
   add_library(soplex ALIAS soplex::soplex)
endif()

```

---

Thank you, that works fine

---

That does not work as expected:

```
CMake Error at /home/conan/w/prod/BuildSingleReference/cci-6f2ef971/recipes/soplex/all/test_package/CMakeLists.txt:8 (add_library):
  add_library cannot create ALIAS target "soplex" because target
  "soplex::soplex" is imported but not globally visible.
```

https://github.com/conan-io/conan-center-index/pull/16017#issuecomment-1504815795

---

https://github.com/conan-io/conan/issues/12654 🤔 guess we need more magic https://discourse.cmake.org/t/alias-and-not-globally-visible-targets/6305/2

```suggestion
find_package(soplex REQUIRED CONFIG)

#workaround v1 legacy generators always having namespaces
add_library(soplex INTERFACE IMPORTED)
set_target_properties(soplex PROPERTIES INTERFACE_LINK_LIBRARIES soplex::soplex)
```

---

Thank you very much for your help! That fixed the problem. Only the linking problem with gmp remains, which I do not understand, as gmp is set as a requirement